File: gb-angular-frontend/src/app/interfaces/package.ts
import {PackageItem} from './package-item';

export interface Package {
  package_id?: number;
  package_name: string;
  creation_date: string;
  description: string;
  quantity: number;
  regular_price: number;
  sale_price: number;
  product_id: number;
  product_title: string;
  product_item_id: number;
  item_name: string;
  package_grp_id: number;
  status: string;
  popularity: number;
  likes: number;
  wished: string;
  grpbuy_order_count: number;
  package_item_list: PackageItem[];
}

File: gb-spring-backend/src/main/java/com/egroupbuy/model/Package.java
package com.egroupbuy.model;

import lombok.*;

import java.sql.Date;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Package {
    int package_id;
    String package_name;
    Date creation_date;
    String description;
    int quantity;
    double regular_price;
    double sale_price;
    int product_id;
    String product_title;
    int product_item_id;
    String item_name;
    int package_grp_id;
    String status;
    int popularity;
    int likes;
    String wished;
    int grpbuy_order_count;
    List<PackageItem> package_item_list;
}
